### PR TITLE
ci: add a new job for checking installation on Ubuntu

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

**Comment by komainu8** [comment]
Mroonga 11.02のUbuntu向けのパッケージのインストールに失敗する。

```
$ sudo apt install -y -V mariadb-server-mroonga
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 mariadb-server-mroonga : Depends: mariadb-server-10.3-mroonga (= 11.02-1.ubuntu20.04.1) but it is not going to be installed
                          Depends: mariadb-server (= 1:10.3.25-0ubuntu0.20.04.1) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

**Comment by komainu8** [comment]
`mariadb-server (= 1:10.3.25-0ubuntu0.20.04.1)`を要求しているが、Ubuntu 20.04のmariadb-serverは、`10.3.29-0ubuntu0.20.04.1`なので、依存パッケージのバージョンの指定がおかしい。

**Comment by komainu8** [comment]
`mariadb-serve`のバージョンを指定しているのは、`packages/mariadb-server-10.3-mroonga/debina/control.in`
この中で `mariadb-server-10.3 (= @MYSQL_VERSION@)` と定義されているので、`@MYSQL_VERSION@`の値を入れているところを直せば良さそう。

File: .github/workflows/install.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Install to Ubuntu
+on:
+  push:
+    branches:
+      - "*"
+    tags:
+      - "*"
```
Comment (kou) [inline .github/workflows/install.yml:7]: We don't need to run this on `push`:

```suggestion
```

File: .github/workflows/install.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        0 0 * * *
+jobs:
+  install:
+    name: Install to Ubuntu
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-18.04
+            package-type: apt
```
Comment (kou) [inline .github/workflows/install.yml:19]: Do we need this?
It seems that we can always use `apt`.

File: packages/apt/install_to_ubuntu.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -exu
+
+package=$1
+
+sudo apt update
+sudo apt-get install -y -V software-properties-common lsb-release
```
Comment (kou) [inline packages/apt/install_to_ubuntu.sh:8]: Could you use `apt` instead of `apt-get`?

File: .github/workflows/install.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      - uses: actions/checkout@v2
+      - name: Run VM
+        run: |
+          vagrant up ${{ matrix.os }}
+      - name: Install Mroonga
+        run: |
+          vagrant \
+            ssh ${{ matrix.os }} \
+            -- \
+            /vagrant/packages/${{ matrix.package-type }}/install_to_ubuntu.sh \
```
Comment (kou) [inline .github/workflows/install.yml:38]: How about `install_test.sh` instead of `install_to_ubuntu.sh`?

**Review by kou — CHANGES_REQUESTED** [review]

File: .github/workflows/install.yml (hunk 1 above)
Comment (komainu8) [reply .github/workflows/install.yml]: Ah... I see.
I've fixed it.

File: .github/workflows/install.yml (hunk 2 above)
Comment (komainu8) [reply .github/workflows/install.yml:14]: Oh... This is needless.
I've fixed it.

File: .github/workflows/install.yml (hunk 4 above)
Comment (komainu8) [reply .github/workflows/install.yml:30]: I agree with you.
I've fixed it.

File: packages/apt/install_to_ubuntu.sh (hunk 3 above)
Comment (komainu8) [reply packages/apt/install_to_ubuntu.sh]: Ah...
I've fixed it.

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I've fixed them. Could you review them again?

File: .github/workflows/install.yml (hunk 4 above)
Comment (kou) [reply .github/workflows/install.yml:30]:
```suggestion
            /vagrant/packages/apt/install_test.sh \
```

File: .github/workflows/install.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        0 0 * * *
+jobs:
+  install:
+    name: Install to Ubuntu
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-18.04
+            package: mysql-server
```
Comment (kou) [inline .github/workflows/install.yml:14]: We'll change package name by the next release:

```suggestion
            package: mysql-5.7
```

File: .github/workflows/install.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+          - os: ubuntu-20.04
+            package: mariadb-server
```
Comment (kou) [inline .github/workflows/install.yml:16]: We'll change package name by the next release:

```suggestion
            package: mariadb
```

File: .github/workflows/install.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+          - os: ubuntu-20.10
+            package: mariadb-server
```
Comment (kou) [inline .github/workflows/install.yml:18]: We'll change package name by the next release:

```suggestion
            package: mariadb
```

**Comment by komainu8** [comment]
I've fixed package names at https://github.com/mroonga/mroonga/pull/348/commits/740eda20e97a1a43fac5e5e58ec0ed887059d424.

**Comment by kou** [comment]
Let's try.

**Pull request merged**